### PR TITLE
add some points

### PR DESCRIPTION
**Pull request by mshustov**

### CODE REVIEW

File: spa-checklist.md [hunk 1]
```diff
@@ -30,20 +30,34 @@ This is a checklist for browser-based Web App **without the backend** or with a
 
 ## Versioning
 
- - [ ] My app has a cache-busting implemented 
+ - [ ] My app has a assets cache-busting implemented
```
Comment (mshustov) [inline spa-checklist.md]: could be moved into `Assets` section?

Comment (kirillDanshin) [reply spa-checklist.md]: why you replace this line?

File: spa-checklist.md [hunk 2]
```diff
@@ -30,20 +30,34 @@ This is a checklist for browser-based Web App **without the backend** or with a
...
+
+## Assets
+
+ - [ ] Have a 404-page
+ - [ ] Have a maintenance page
```
Comment (mr-mig) [inline spa-checklist.md]: What is a "maintenance page"?

File: spa-checklist.md (hunk 1 above)
Comment (mr-mig) [reply spa-checklist.md]: Please, remove an article in the sentence: 
"My app has assets cache-busting implemented"

Comment (mr-mig) [reply spa-checklist.md]: I'd leave this line here, as asset cache-busting is essential for sane versioning

File: spa-checklist.md [hunk 3]
```diff
@@ -30,20 +30,34 @@ This is a checklist for browser-based Web App **without the backend** or with a
...
     - [ ] [PageSpeed](https://developers.google.com/speed/pagespeed/) 
     - [ ] [YSlow](http://yslow.org/)
     - [ ] Chrome Dev Tools Audit
 
 ## Debugging
 
- - [ ] minified JavaScript files contain URL for source maps
+ - [ ] Minified JavaScript files contain URL for source maps
+ - [ ] Thrown exceptions are handled and passed to storage ([Sentry](https://sentry.io/), [Track.js](https://trackjs.com/), etc)
+ - [ ] Error storage process sourcemaped files
```
Comment (mr-mig) [inline spa-checklist.md:54]: Great additions overall, thank you 👍

File: spa-checklist.md (hunk 2 above)
Comment (kirillDanshin) [reply spa-checklist.md]: @mr-mig "Sorry, we'll back ASAP, you can now go for a coffee-break and refresh this page than"

File: spa-checklist.md (hunk 1 above)
Comment (kirillDanshin) [reply spa-checklist.md]: @restrry @mr-mig I think we need to leave booth lines or replace this with
"My app has a cache-busting implemented (assets, object cache etc.)"

Comment (mr-mig) [reply spa-checklist.md]: > "My app has a cache-busting implemented (assets, object cache etc.)"
:+1:

File: spa-checklist.md (hunk 2 above)
Comment (mr-mig) [reply spa-checklist.md]: Could you please add it to the description?

**Comment by mshustov** [comment]
@kirillDanshin @mr-mig addressed, pls have a look

File: spa-checklist.md [hunk 4]
```diff
@@ -30,20 +30,34 @@ This is a checklist for browser-based Web App **without the backend** or with a
 
 ## Versioning
 
- - [ ] My app has a cache-busting implemented 
+ - [ ] My app has a cache-busting implemented (assets, object cache etc.)
```
Comment (kirillDanshin) [inline spa-checklist.md:33]: 👍

File: spa-checklist.md [hunk 5]
```diff
@@ -30,20 +30,34 @@ This is a checklist for browser-based Web App **without the backend** or with a
...
+ - [ ] Have a [maintenance page](https://www.smashingmagazine.com/2009/06/effective-maintenance-pages-examples-and-best-practices/)
+ - [ ] Images support HDPI screens (Retina, etc)
 
 
 ## Testing
-  - [ ] My app does not have memory leaks
-  - [ ] My app is passing performance tests with high grades
+
+ - [ ] My app does not have memory leaks
+ - [ ] My app is passing performance tests with high grades
```
Comment (kirillDanshin) [inline spa-checklist.md:45]: we need to add
`- [ ] My app is loading in less than 3 seconds`

File: spa-checklist.md [hunk 6]
```diff
@@ -30,20 +30,34 @@ This is a checklist for browser-based Web App **without the backend** or with a
...
+ - [ ] Thrown exceptions are handled and passed to storage ([Sentry](https://sentry.io/), [Track.js](https://trackjs.com/), etc)
+ - [ ] Error storage process sourcemaped files
 
 ## Tracking
 
  - [ ] My app gathers metrics about usage behaviour
+
+## Security
+ - [ ] I have audited my system against:
+    - [ ] OWASP Top 10 Vulnerabilities
```
Comment (kirillDanshin) [inline spa-checklist.md:62]: maybe add a link? https://www.owasp.org/index.php/Category:OWASP_Top_Ten_Project or https://www.owasp.org/index.php/OWASP_Top_Ten_Cheat_Sheet

**Review by kirillDanshin — CHANGES_REQUESTED** [review]

**Comment by mr-mig** [comment]
Merged. Thanks, great additions!

**Comment by mr-mig** [comment]
@kirillDanshin I'll add your comments manually

File: spa-checklist.md (hunk 5 above)
Comment (mshustov) [reply spa-checklist.md:45]: Probably makes sense to add link to [RAIL model](https://developers.google.com/web/fundamentals/performance/rail) in that case

**Pull request merged**